File: frontend-api/src/test/java/no/nav/melosys/tjenester/gui/dto/dokumentarkiv/JournalpostInfoDtoTest.java
package no.nav.melosys.tjenester.gui.dto.dokumentarkiv;

import java.time.Instant;

import no.nav.melosys.domain.arkiv.ArkivDokument;
import no.nav.melosys.domain.arkiv.Journalpost;
import no.nav.melosys.domain.arkiv.Journalposttype;
import no.nav.melosys.domain.kodeverk.Mottaksretning;
import org.junit.Test;

import static org.assertj.core.api.Assertions.assertThat;

public class JournalpostInfoDtoTest {

    @Test
    public void av() {
        String journalpostID = "journalpostID";
        Journalpost journalpost = new Journalpost(journalpostID);
        journalpost.setJournalposttype(Journalposttype.INN);
        Instant nå = Instant.now();
        journalpost.setForsendelseMottatt(nå);
        journalpost.setForsendelseJournalfoert(nå);
        ArkivDokument hoveddokument = new ArkivDokument();
        String hovedTittel = "tittel";
        hoveddokument.setTittel(hovedTittel);
        hoveddokument.setDokumentId("1");
        journalpost.setHoveddokument(hoveddokument);
        String partNavn = "part navn";
        journalpost.setKorrespondansepartNavn(partNavn);

        ArkivDokument v = new ArkivDokument();
        String vedleggsTittel = "tittelV";
        v.setTittel(vedleggsTittel);
        v.setDokumentId("2");
        journalpost.getVedleggListe().add(v);

        JournalpostInfoDto dto = JournalpostInfoDto.av(journalpost);

        assertThat(dto.journalpostID).isEqualTo(journalpostID);
        assertThat(dto.mottaksretning).isEqualTo(Mottaksretning.INN);
        assertThat(dto.mottattDato).isNotNull();
        assertThat(dto.journalforingDato).isNotNull();
        assertThat(dto.avsenderEllerMottaker).isEqualTo(partNavn);
        assertThat(dto.hoveddokument.dokumentID).isEqualTo("1");
        assertThat(dto.hoveddokument.tittel).isEqualTo(hovedTittel);
        assertThat(dto.vedlegg.get(0).dokumentID).isEqualTo("2");
        assertThat(dto.vedlegg.get(0).tittel).isEqualTo(vedleggsTittel);
    }
}

File: frontend-api/src/test/kotlin/no/nav/melosys/tjenester/gui/dto/dokumentarkiv/JournalpostInfoDtoTest.kt
package no.nav.melosys.tjenester.gui.dto.dokumentarkiv

import io.kotest.matchers.shouldBe
import io.kotest.matchers.shouldNotBe
import no.nav.melosys.domain.arkiv.ArkivDokument
import no.nav.melosys.domain.arkiv.Journalpost
import no.nav.melosys.domain.arkiv.Journalposttype
import no.nav.melosys.domain.kodeverk.Mottaksretning
import org.junit.jupiter.api.Test
import java.time.Instant

class JournalpostInfoDtoTest {

    @Test
    fun `skal konvertere journalpost til journalpostInfoDto`() {
        val journalpostID = "journalpostID"
        val hovedTittel = "tittel"
        val partNavn = "part navn"
        val vedleggsTittel = "tittelV"
        val journalpost = Journalpost(journalpostID).apply {
            journalposttype = Journalposttype.INN
            val nå = Instant.now()
            forsendelseMottatt = nå
            forsendelseJournalfoert = nå
            hoveddokument = ArkivDokument().apply {
                tittel = hovedTittel
                dokumentId = "1"
            }
            korrespondansepartNavn = partNavn
            vedleggListe.add(ArkivDokument().apply {
                tittel = vedleggsTittel
                dokumentId = "2"
            })
        }


        val dto = JournalpostInfoDto.av(journalpost)


        dto.run {
            this.journalpostID shouldBe journalpostID
            mottaksretning shouldBe Mottaksretning.INN
            mottattDato shouldNotBe null
            journalforingDato shouldNotBe null
            avsenderEllerMottaker shouldBe partNavn
            hoveddokument.run {
                dokumentID shouldBe "1"
                tittel shouldBe hovedTittel
            }
            vedlegg[0].run {
                dokumentID shouldBe "2"
                tittel shouldBe vedleggsTittel
            }
        }
    }
}